File: 240205/줄 세우기/line-up-students.java
import java.io.*;
import java.util.*;

class Student implements Comparable<Student> {
    int number, height, weight;
    public Student(int number, int height, int weight) {
        this.number = number;
        this.height = height;
        this.weight = weight;
    }

    @Override
    public int compareTo(Student student) {
        if (this.height != student.height) {
            return student.height - this.height;
        }
        if (this.weight != student.weight) {
            return student.weight - this.weight;
        }
        return this.number - student.number;
    }

    public void print() {
        System.out.println(String.format("%d %d %d", this.height, this.weight, this.number));
    }
}

public class Main {
    public static int inputLength;
    public static List<Student> students = new ArrayList<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        inputLength = Integer.parseInt(br.readLine());
        for (int i = 0; i < inputLength; i++) {
            int[] inputs = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            int number = i + 1;
            int height = inputs[0];
            int weight = inputs[1];
            students.add(new Student(number, height, weight));
        }

        Collections.sort(students);
        for (Student student : students) {
            student.print();
        }
    }
}

File: 240317/줄 세우기/line-up-students.py
class Student:
    def __init__(self, height, weight, number):
        self.height = height
        self.weight = weight
        self.number = number

n = int(input())
students_info = [list(map(int, input().split())) for _ in range(n)]
students = [Student(height, weight, number) for number, (height, weight) in enumerate(students_info, start=1)]
    

students.sort(key=lambda ele: (-ele.height, -ele.weight, ele.number))
for ele in students:
    print(ele.height, ele.weight, ele.number)